Добавлены тесты на проверку ручек

grayyyyy456 opened the pull request.

grayyyyy456 committed:
Добавлены тесты на проверку ручек

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,25 @@
+import string
+
+# генерация рандомного логина, пароля и имени
+def register_new_courier_and_return_login_password():
```
Нужно исправить: модуль conftest предназначен только для фикстур и хуков.

ShamilovR commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,36 @@
+import allure
+
+class TestCreateCourier:
+    @allure.title("Проверка на создание курьера")
+    def test_create_courier(self):
+        data_create_courier = register_new_courier_and_return_login_password()
+        response = requests.post(creating_courier_api, json=data_create_courier)
+        assert response.status_code == 201 and response.text == '{"ok":true}'
+
+    @allure.title("Проверка на ошибку, при попытке создать двух одинаковых курьера")
+    def test_create_two_identical_couriers(self):
+        data_create_courier = register_new_courier_and_return_login_password()
+        response_1 = requests.post(creating_courier_api, json=data_create_courier)
+        response_2 = requests.post(creating_courier_api, json=data_create_courier)
+        assert response_2.status_code == 409 and 'Этот логин уже используется. Попробуйте другой.' in response_2.text
```
Нужно исправить: отсутствует функционал удаления курьера после выполнения тестового метода. Удаление должно быть реализовано в рамках фикстур.

ShamilovR replied:
Можно лучше: здесь и далее: тексты ответов от сервера следует выносить в дата модули. В случае изменении текста ответа от сервера, не придется править в каждом тесте отдельно. Достаточно будет поправить лишь в одном месте - дата модуле.

ShamilovR commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,36 @@
+    @allure.title("Проверка на создание курьера")
+    def test_create_courier(self):
+        data_create_courier = register_new_courier_and_return_login_password()
+        response = requests.post(creating_courier_api, json=data_create_courier)
```
Можно лучше: можно вынести вызовы апи методов в отельные классы. И объединять их по сущности. Такая реализация позволит удобно переиспользовать эти методы. (Например, можно разделить на классы Курьер и Заказ)

grayyyyy456 committed:
Генерация рандомных данных была перенесена в файл helper.py
Добавлена фикстура для удаления курьера
Изменены тесты test_create_two_identical_couriers, test_create_courier

grayyyyy456 committed:
статус коды были вынесены в data файл и вызываются в тестах

grayyyyy456 committed:
тексты ответов были вынесены в data файл и вызываются в тестах

grayyyyy456 committed:
тексты ответов DELETE были вынесены в data файл и вызываются в тестах

grayyyyy456 committed:
добавлен файл api_methods.py для api методов и вызываются в тестах

grayyyyy456 committed:
last commit

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,18 @@
+def courier():
+    data_create_courier = register_new_courier_and_return_login_password()
+    response = ApiMethodsCourier.create_courier(data_create_courier)
+    assert response.status_code == StatusCode.status_code_201 and response.json() == ResponseTextCreateCourier.text_status_code_201
```
Можно лучше: проверки в фикстурах избыточны. Они должны производиться в рамках тестовых методов

ShamilovR commented on tests/conftest.py:
```diff
@@ -0,0 +1,18 @@
+import pytest
+from helpers import register_new_courier_and_return_login_password
+from data import ResponseTextCreateCourier, StatusCode, ResponseTextDeleteCourier
+from api_methods import ApiMethodsCourier
+
+
+@pytest.fixture
+def courier():
+    data_create_courier = register_new_courier_and_return_login_password()
+    response = ApiMethodsCourier.create_courier(data_create_courier)
+    assert response.status_code == StatusCode.status_code_201 and response.json() == ResponseTextCreateCourier.text_status_code_201
+    yield data_create_courier
+    login = data_create_courier['login']
+    password = data_create_courier['password']
+    id_response = ApiMethodsCourier.login_courier(login, password)
+    courier_id = id_response.json().get('id')
+    delete_response = ApiMethodsCourier.delete_courier(courier_id)
```
Можно лучше: полностью дублирует корневой conftest. От одного из них можно избавиться

ShamilovR commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,34 @@
+from data import DataCourier, StatusCode, ResponseTextCreateCourier
+import allure
+from api_methods import ApiMethodsCourier
+from conftest import courier
```
Нужно исправить: фикстуры не нужно импортировать. Если они лежат в conftest то pytest их видит без импорта

grayyyyy456 committed:
Убрал второй файл conftest.py
Убрал проверки assert в фикстуре
Убрал импорт фикстуры

grayyyyy456 committed:
last
